src/components/media/MediasSelectModal.jsx

```diff
@@ -52,35 +57,45 @@ const MediasSelectModal = ({
   setQueryParams,
 }) => {
   const { params } = useRouteMatch()
-  const { siteName } = params
+  const { siteName, fileName } = params
 
   const { mediaRoom, mediaDirectoryName } = queryParams
 
   const [curPage, setCurPage] = usePaginate()
 
+  const [mediaFolderSubdirectories, setMediaFolderSubdirectories] = useState([])
+  const [mediaFolderFiles, setMediaFolderFiles] = useState([])
+  const [total, setTotal] = useState()
+
   const {
-    data: { directories: mediaFolderSubdirectories },
+    data: listMediaDirectoriesData,
     isLoading: isListMediaFolderSubdirectoriesLoading,
-  } = useListMediaFolderSubdirectories(
-    {
-      ...queryParams,
-    },
-    { initialData: { directories: [] } }
-  )
+  } = useListMediaFolderSubdirectories({
+    ...queryParams,
+  })
 
   const {
-    data: { files: mediaFolderFiles, total },
+    data: listMediaFilesData,
     isLoading: isListMediaFilesLoading,
-  } = useListMediaFolderFiles(
-    {
-      ...queryParams,
-      // NOTE: Subtracting 1 here because `usePaginate`
-      // returns an index with 1 offset
-      curPage: curPage - 1,
-      limit: MEDIA_PAGINATION_SIZE,
-    },
-    { initialData: { files: [] } }
-  )
+  } = useListMediaFolderFiles({
+    ...queryParams,
+    // NOTE: Subtracting 1 here because `usePaginate`
+    // returns an index with 1 offset
+    curPage: curPage - 1,
+    limit: MEDIA_PAGINATION_SIZE,
+  })
+
+  useEffect(() => {
+    if (listMediaDirectoriesData)
+      setMediaFolderSubdirectories(listMediaDirectoriesData.directories)
+  }, [listMediaDirectoriesData])
+
+  useEffect(() => {
+    if (listMediaFilesData) {
+      setMediaFolderFiles(listMediaFilesData.files)
+      setTotal(listMediaFilesData.total)
+    }
+  }, [listMediaFilesData])
```

Comment:
not very sure why this is required but in alex's absence, will approve for now cos it doesn't impact functionality

Reply:
This was blocking the image display until after all images were loaded, since we were defaulting the return initially to `[]`  - moving it into it's own separate useEffect allows us to track the load progress independently